src/deroff.rs

```diff
@@ -1028,17 +1044,98 @@ impl Deroffer {
         }
     }
 
-    fn do_line(&self, s: &str) -> bool {
-        match s.chars().nth(0) {
-            Some('.') | Some('\'') => !request_or_macro(s),
-            Some(c) => {
+    fn do_tbl(&mut self) -> bool {
+        match self.tblstate {
+            TblState::Options => {
+                while !self.s.is_empty() && !"\n;".contains(self.str_at(0)) {
+                    self.skip_leading_whitespace();
+
+                    if !self.str_at(0).chars().all(|c| c.is_alphabetic()) {
+                        self.skip_char(1);
+                    } else {
+                        let mut option = self.s.to_owned();
+                        let mut arg = String::new();
+
+                        let mut idx: usize = 0;
+
+                        while option
+                            .get(idx..=idx)
+                            .map_or(false, |s| s.chars().all(|c| c.is_alphabetic()))
+                        {
+                            idx += 1;
+                        }
+
+                        if option.get(idx..=idx) == Some("(") {
+                            option = option.get(..idx).unwrap_or_default().to_owned();
+                            self.s = self.s.get(idx + 1..).unwrap_or_default().to_owned();
+                            arg = self.s.clone();
+                        } else {
+                            self.s.clear();
+                        }
+
+                        if !arg.is_empty() {
+                            if let Some(idx) = arg.find("(") {
+                                arg = arg.get(..idx).unwrap_or_default().to_owned();
+                                self.s.drain(..=idx);
+                            }
+                        } else {
+                            // This was commented out in the python, so it is here too :)
+                            // self.skip_char(1);
+                        }
+
+                        if option.to_lowercase() == "tab" {
+                            self.tblTab = arg.get(0..1).unwrap_or_default().to_owned();
+                        }
```

Comment:
Let me bend this a bit for you, needs some review. Heh, this one does no allocation.
```suggestion
                        // find first non-alphabetic character
                        match self.s.chars().position(|c| !c.is_alphabetic()) {
                            Some(idx) if self.s.get(idx..=idx) == Some("(") => {
                                // self.s -> option '(' arg '(' rest
                                let option = self.s.split_at(idx).0;
                                let mut iter = self.s.split_at(idx + 1).1.splitn(2, '(');
                                let arg = iter.next().unwrap_or_default();
                                let rest = iter.next().unwrap_or_default();

                                if option.to_lowercase() == "tab" {
                                    self.tblTab = arg.get(0..1).unwrap_or_default().to_owned();
                                }

                                // Likely faster alternative to self.s = rest
                                self.s.drain(self.s.len() - rest.len()..);
                            }
                            _ => self.s.clear(),
                        }
```